Cheatsheet.md

```diff
@@ -0,0 +1,504 @@
+
+**Return value:**
+ The array element value if any of the elements in the array pass the test, otherwise it returns 'undefined'.
+
+**Example:**
+ ```  
+ let array = [2, 7, 8, 9]; 
+ const found = array.find(element=>element > 4); 
+ // Printing desired values. 
+ console.log(found); 
+```
+*Output:*
+    
+    > 7
+
+### [10. indexOf](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/indexOf)
+
+**Description:**
+returns the position of the first occurrence of a specified value in a string.
+
+**Syntax:**
+ * _string.indexOf(searchvalue, start)_
+
+| Parameter | Description |
+| :-------- | :---------- |
+| searchvalue | Required. The string to search for |
+| start	      | Optional. <br>Default 0. At which position to start the search.|
+
+**Return value:**
+Number, representing the position where the specified searchvalue occurs for the first time, or -1 if it never occurs.
+
+**Example:**
+```
+console.log('Departed Train'.indexOf('Train'));
+```
+*Output:*
+    
+    9
+    
+## Commonly used functions on [string](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String) 
+
+### [1. substring](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/String/substring)
+
+**Description:**
+Extracts the characters in a string between "start" and "end", not including "end" itself.
+
+**Syntax:**
+ * _string.substring(start, end)_
+
+| Parameter | Description |
+| :-------- | :---------- |
+|start	| Required. The position where to start the extraction. First character is at index 0 |
+| end	| Optional. The position (up to, but not including) where to end the extraction. If omitted, it extracts the rest of the string|
+
+**Return value:**
+New string containing the extracted characters.
+
+**Example:**
+```
+// Taking a string as variable 
+const string = "geeksforgeeks"; 
+a = string.substring(0, 4) 
+b = string.substring(1, 6) 
+c = string.substring(5) 
+d = string.substring(0) 
+// Printing new string which are 
+// the part of the given string 
+console.log(a); 
+console.log(b); 
+console.log(c); 
+console.log(d); 
+ ```
+*Output:*
+    
+    geek
+    eeksf
+    forgeeks
+    geeksforgeeks
+
+### [2. split](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/split)
+
+**Description:**
+Used to split a string into an array of substrings, and returns the new array.
+
+**Syntax:**
+ * _string.split(separator, limit)_
+
+| Parameter | Description |
+| :-------- | :---------- |
+|separator	| Optional. Specifies the character, or the regular expression, to use for splitting the string. If omitted, the entire string will be returned (an array with only one item)|
+| limit	| Optional. An integer that specifies the number of splits, items after the split limit will not be included in the array |
+
+**Return value:**
+Array containing splitted values.
+
+**Example:**
+```
+let str = 'It iS a 5r&e@@t Day.'

```

Comment:
please use different test data, as leet-speak isn't comfortable for beginners if they don't know about it